### PR TITLE
refactoring auth, agent and auto role selection tests to use Cluster object

### PR DESCRIPTION
**Pull request by nshidlin**

### CODE REVIEW

File: discovery-infra/tests/test_agent.py [hunk 1]
```diff
@@ -11,26 +11,26 @@ def test_kill_agent(self, api_client, nodes, cluster):
...
-        cluster_id = cluster().id
+        new_cluster = cluster()
         # Generate and download cluster ISO
-        self.generate_and_download_image(cluster_id=cluster_id, api_client=api_client)
+        new_cluster.generate_and_download_image()
         # Boot nodes into ISO
-        nodes.start_all_nodes()
-        test_host = nodes.nodes[0]
+        nodes.start_all()
+        test_node = nodes.nodes[0]
```
Comment (lalon4) [inline discovery-infra/tests/test_agent.py:19]: use nodes.get_random_node()

**Comment by osherdp** [comment]
LGTM

File: discovery-infra/tests/test_auto_role_selection.py [hunk 2]
```diff
@@ -7,40 +7,28 @@
...
-        self.setup_hosts(cluster_id=cluster_id,
-                         api_client=api_client,
-                         nodes=nodes)
-        self.set_network_params(cluster_id=cluster_id,
-                                api_client=api_client,
-                                controller=nodes.controller)
+        new_cluster.setup_nodes(nodes)
+        new_cluster.set_network_params(nodes.controller)
+        new_cluster.wait_for_ready_to_install()
+        new_cluster.start_install()
```
Comment (lalon4) [inline discovery-infra/tests/test_auto_role_selection.py:15]: The start install on Base is also wait_until_cluster_starts_installation.
What happen here? maybe need to add wait for installing in progress?

Comment (nshidlin) [reply discovery-infra/tests/test_auto_role_selection.py:15]: changed

File: discovery-infra/tests/test_agent.py (hunk 1 above)
Comment (nshidlin) [reply discovery-infra/tests/test_agent.py]: changed

File: discovery-infra/test_infra/helper_classes/nodes.py [hunk 3]
```diff
@@ -101,10 +101,11 @@ def get_bootstrap_node(self, cluster) -> Node:
 
     def create_nodes_cluster_hosts_mapping(self, cluster):
         node_mapping_dict = {}
-        for cluster_host_object in cluster.get_nodes():
+        for cluster_host_object in cluster.get_hosts():
             inventory = json.loads(cluster_host_object["inventory"])
```
Comment (tsorya) [inline discovery-infra/test_infra/helper_classes/nodes.py:105]: i think we can move logic getting hostname to function

Comment (tsorya) [reply discovery-infra/test_infra/helper_classes/nodes.py:105]: or i think we even have it

Comment (nshidlin) [reply discovery-infra/test_infra/helper_classes/nodes.py]: changed

**Pull request merged**